test_env_setup_util/libs/model.py

```diff
@@ -0,0 +1,122 @@
+from pydantic import BaseModel, model_validator, validator, Discriminator, Tag
+from typing import Annotated, Literal, Optional, Union
+
+
+class BaseAction(BaseModel):
+    """Base model for all actions, used for discriminated union."""
+
+    ignore_error: Optional[bool] = False
+    bypass_condition: Optional[str] = ""
+
+
+class InstallSnapAction(BaseAction):
+    action: Literal["install_snap"]
+    name: str
+    track: Optional[str] = "latest"
+    risk: Optional[str] = "stable"
+    branch: Optional[str] = ""
+    revision: Optional[str] = ""
+    mode: Optional[str] = ""
+    post_commands: Optional[str] = ""
+
+    @validator("mode")
```

Comment:
I think this is deprecated: https://docs.pydantic.dev/latest/migration/#validator-and-root_validator-are-deprecated